docs/sources/alerting/rules.md

```diff
@@ -51,7 +51,9 @@ Here you can specify the name of the alert rule and how often the scheduler shou
 
 ### For
 
-> This setting is available in Grafana 5.4 and above.
+> **Important note regarding No Data:**
+>
+> When configuring an alert to use `For` and `If no data or all values are null` set to `No Data` the triggering of `No Data` will trigger instantly and not take `For` into consideration. This may also result in that an OK notification not being sent if alert transitions from `No Data -> Pending -> OK`.
```

Comment:
```suggestion
> Do not use `For` with the `If no data or all values are null` setting set to `No Data`. The triggering of `No Data` will trigger instantly and not take `For` into consideration. This may also result in that an OK notification not being sent if alert transitions from `No Data -> Pending -> OK`.
```